### PR TITLE
dillo: 3.0.5-unstable-2021-02-09 -> 3.1.0

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@fgaz
As far as I have seen, `dillong` is outdated and [dillo-plus](https://github.com/crossbowerbt/dillo-plus/) picked code from them, making it a more "up-to-date version". I was thinking about opening a new commit for this.

**Comment by fgaz** [comment]
Looks like I looked into this a few months ago and reached the same conclusion, here's a note from my wiki:

> add/replace dillo forks
> * dillo → https://github.com/dillo-browser/dillo 
> * dillong → https://github.com/crossbowerbt/dillo-plus

so go ahead with the dillong → dillo-plus pr.

Unfortunately it seems like the dillo-browser fork [did not receive the original authors' blessing to use the same name](https://github.com/dillo-browser/dillo/issues/53), but original dillo is long dead so I feel like switching to this fork is ok.

**Comment by rodarima** [comment]
> Looks like I looked into this a few months ago and reached the same conclusion, here's a note from my wiki:
> 
> > add/replace dillo forks
> > 
> > * dillo → https://github.com/dillo-browser/dillo
> > * dillong → https://github.com/crossbowerbt/dillo-plus
> 
> so go ahead with the dillong → dillo-plus pr.
> 
> Unfortunately it seems like the dillo-browser fork [did not receive the original authors' blessing to use the same name](https://github.com/dillo-browser/dillo/issues/53), but original dillo is long dead so I feel like switching to this fork is ok.

AFAIK bkil has not contributed to the dillo project nor any fork of it. Please read the relase notes and changelog to see more details and see who contributed what: https://dillo-browser.github.io/latest.html

We keep trying to contact Jorge and [other core developers](https://dillo-browser.github.io/old/D_authors.html). 

Some of them are aware, here is Johannes reply:

> Date: Sat, 30 Dec 2023 10:31:12 +0100 (GMT+01:00)
> From: Johannes Hofmann
> To: Rodrigo Arias Mallo
> Cc: Jorge Arellano Cid, Andreas Kemnade, Sebastian Geerken, corvid, Luca Rota 
> Subject: Re: Resurrecting Dillo
> 
> Hi Rodrigo,
> 
> I didn't hear from Jorge for quite some time. I'm happy to help you to get the
> latest sources but it may take some time.
> 
> Best Regards,
> Johannes
> 
> 30.12.2023 03:44:13 Rodrigo Arias Mallo <rodarima@gmail.com>:
> 
> > Hi,
> >
> > I don't want Dillo to die, so I'm trying to resurrect it here:
> >
> > https://github.com/dillo-browser/dillo
> >
> > Since the dead of dillo.org I cannot longer contact any of the
> > developers via email or the mailing list. Is any of you still in contact
> > with Jorge or other core developers or can provide a way to reach them?
> >
> > Also, reading the mercurial logs I see references to a test suite like
> > "http://www.dillo.org/test/4648/test-suite.v1.txt" which would be nice
> > to recover, if you still have access to the server.
> >
> > I'm guessing Dillo would reach 25 years soon, so it would be nice to
> > celebrate :-)
> >
> > Best regards,
> > Rodrigo.

Regarding the PR, thanks for updating it so quickly! You can also link Dillo with LibreSSL in case you want to add it to the TODO line.

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 309188` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>dillo</li>
    <li>dillo.doc</li>
    <li>dillo.man</li>
    <li>dillong</li>
    <li>dillong.doc</li>
    <li>dillong.man</li>
  </ul>
</details>

**Review by lucasew — APPROVED** [review]
It seems to be working as expected.

Most sites can't be properly used and the CSS works buggy but no crashes or weird errors.

**Comment by AndersonTorres** [comment]
> Regarding the PR, thanks for updating it so quickly! You can also link Dillo with LibreSSL in case you want to add it to the TODO line.

I was reading a Telegram bot that replicates Hackernews posts, and the name `dillo` hit my attention.
Many thanks for revamping Dillo!

https://news.ycombinator.com/item?id=38847613

**Review by rodarima — APPROVED** [review]

File: pkgs/by-name/di/dillong/package.nix [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-, mbedtls_2
+{
+  lib,
+  autoreconfHook,
+  fetchFromGitHub,
+  fltk,
+  mbedtls_2,
+  pkg-config,
+  stdenv,
+  which,
```
Comment (fgaz) [inline pkgs/by-name/di/dillong/package.nix:9]: Nitpick: gratuitous reformatting

**Review by fgaz — APPROVED** [review]
Looks good.

@rodarima thanks for the info, good to know that an attempt was made

**Comment by fgaz** [comment]
@AndersonTorres do you still plan to open a pr adding dillo-plus?

**Comment by AndersonTorres** [comment]
Yes, I am interested. I am studying their building system - a typical Makefile pointing to `/usr/lib` things...

**Pull request merged**